src/Rules/UnusedPublicClassMethodRule.php

```diff
@@ -115,4 +123,25 @@ private function isUsedClassMethod(
         $methodReference = $className . '::' . $methodName;
         return in_array($methodReference, $completeMethodCallReferences, true);
     }
+
+    private function isTraitMethod(string $className, string $methodName, Scope $scope): bool
+    {
+        if (!$this->reflectionProvider->hasClass($className)) {
+            return false;
+        }
+        $classReflection = $this->reflectionProvider->getClass($className);
+
+        if (!$classReflection->hasMethod($methodName)) {
+            return false;
+        }
+
+        $extendedMethodReflection = $classReflection->getMethod($methodName, $scope);
+        if ($extendedMethodReflection instanceof PhpMethodReflection || $extendedMethodReflection instanceof ResolvedMethodReflection) {
+            return $extendedMethodReflection->getDeclaringTrait() !== null;
+        }
```

Comment:
lets see whether this is correct: https://github.com/phpstan/phpstan/discussions/9186

Reply:
this needs a few hours of work on the phpstan-src side. I think the PR as is is good for now and we should move to a new `ExtendedMethodReflection` based api when it is available in phpstan-src